lms/envs/common.py

```diff
@@ -587,7 +587,7 @@
 # Note: these intentionally greedily grab all chars up to the next slash including any pluses
 # DHM: I really wanted to ensure the separators were the same (+ or /) but all patts I tried had
 # too many inadvertent side effects :-(
-COURSE_KEY_PATTERN = r'(?P<course_key_string>[^/+]+(/|\+)[^/+]+(/|\+)[^/]+)'
+COURSE_KEY_PATTERN = r'(?P<course_key_string>[^/+]+(/|\+)[^/+]+(/|\+)[^/][^?/]*)'
```

Comment:
Why did you change the final capture pattern from using `+` (so that at least a single character is required) to `*` (so that the empty string matches)?